server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -821,22 +1023,23 @@ public void deleteSnapshots(final DeleteSnapshotRequest request, final ActionLis
 
             @Override
             public ClusterState execute(ClusterState currentState) throws Exception {
-                if (snapshotNames.length > 1 && currentState.nodes().getMinNodeVersion().before(MULTI_DELETE_VERSION)) {
+                final Version minNodeVersion = currentState.nodes().getMinNodeVersion();
+                if (snapshotNames.length > 1 && minNodeVersion.before(MULTI_DELETE_VERSION)) {
                     throw new IllegalArgumentException("Deleting multiple snapshots in a single request is only supported in version [ "
-                            + MULTI_DELETE_VERSION + "] but cluster contained node of version [" + currentState.nodes().getMinNodeVersion()
-                            + "]");
+                            + MULTI_DELETE_VERSION + "] but cluster contained node of version [" + minNodeVersion + "]");
                 }
                 final SnapshotsInProgress snapshots = currentState.custom(SnapshotsInProgress.TYPE);
-                final SnapshotsInProgress.Entry snapshotEntry = findInProgressSnapshot(snapshots, snapshotNames, repositoryName);
+                final List<SnapshotsInProgress.Entry> snapshotEntries = findInProgressSnapshots(snapshots, snapshotNames, repositoryName);
                 final List<SnapshotId> snapshotIds = matchingSnapshotIds(
-                        snapshotEntry == null ? null : snapshotEntry.snapshot().getSnapshotId(),
-                        repositoryData, snapshotNames, repositoryName);
-                if (snapshotEntry == null) {
+                        snapshotEntries.stream().map(e -> e.snapshot().getSnapshotId()).collect(Collectors.toList()), repositoryData,
+                        snapshotNames, repositoryName);
+                if (snapshotEntries.isEmpty() || minNodeVersion.onOrAfter(SnapshotsService.FULL_CONCURRENCY_VERSION)) {
                     deleteFromRepoTask =
                             createDeleteStateUpdate(snapshotIds, repositoryName, repositoryData.getGenId(), Priority.NORMAL, listener);
                     return deleteFromRepoTask.execute(currentState);
                 }
-
+                assert snapshotEntries.size() == 1 : "Expected just a single running snapshot but saw " + snapshotEntries;
```

Comment:
why is there only at most one?

Reply:
Because we break out above to the new logic if we have a cluster on or after `FULL_CONCURRENCY_VERSION` and hence never get here unless there  is precisely one running snapshot.